app/com/arpnetworking/metrics/portal/scheduling/impl/UnboundedPeriodicSchedule.java

```diff
@@ -58,10 +58,11 @@ private UnboundedPeriodicSchedule(final Builder builder) {
     @Override
     public Optional<Instant> nextRun(final Optional<Instant> lastRun) {
         final Instant now = _clock.instant();
+
         final Instant start = lastRun
-                .filter(run -> run.compareTo(now) >= 0)
-                .map(run -> run.plus(_fullPeriod)) // Avoid repeating the same period.
-                .orElse(now);
+                .filter(run -> run.isAfter(now))
+                .orElse(now)
+                .plus(_fullPeriod); // Avoid repeating the same period
```

Comment:
Won't this cause the opposite behavior now where there is always an initial delay of `_fullPeriod`?

Reply:
The point here is that when you call `nextRun` it should always be in the future. So I suppose yes, there is a delay.

Here's the before and after, assuming a schedule of every 30 minutes:

Unchanged case, where we're not on a period:
| Field | Value |
| --- | --- |
| Now | 10:01:00 |
| Last Run |  10:00:00 |
| Next Run (before) | 10:30:00 |
| Next Run (after) |  10:30:00 |

Changed case, where we're exactly on a period:
| Field | Value |
| --- | --- |
| Now | 10:00:00 |
| Last Run |  (empty) |
| Next Run (before) | 10:00:00 |
| Next Run (after) |  10:30:00 |

If there were no delay for this, as previously, then the `nextRun` will always round down back to `lastRun` if it were exactly the previous period. e.g. times 10:01, 10:02, 10:03 would all return `nextRun = 10:00:00` unless we add the period.

An alternative would be to add an if-statement to check if we're at a fixed point and intentionally bump it, but I felt that this special-casing made it harder to reason with.

Reply:
Happy to discuss any problems you might see with this. I had run alert evaluation locally and it displayed the expected behavior of running once per minute, starting from the upcoming minute post-startup.